### PR TITLE
create counter of counters

### PR DESCRIPTION
**Pull request by PavelPashkovich**

### CODE REVIEW

File: src/pages/CounterOfCounters/components/CounterItem/index.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import React from "react";
```
Comment (Boobaley) [inline src/pages/CounterOfCounters/components/CounterItem/index.js]: useless import

File: src/pages/CounterOfCounters/components/CountersLayout/index.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  handleRemoveSpecificCounter,
+  handleResetCounters,
+  handleDecrement,
+  handleReset,
+  handleIncrement,
+}) => {
+  const countersSumValue = counters.reduce((sum, counter) => {
+    sum += counter.countValue;
+    return sum;
+  }, 0);
```
Comment (Boobaley) [inline src/pages/CounterOfCounters/components/CountersLayout/index.js]: this must be at the container level

File: src/pages/CounterOfCounters/containers/CounterOfCountersContainer.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  const [counters, setCounters] = useState([]);
+
+  const handleAddCounter = useCallback(() => {
+    setCounters((counters) => {
+      const copyCounters = [...counters];
+      copyCounters.forEach((counter) => {
+        if (counter.countValue % 2 === 0) {
+          ++counter.countValue;
+        }
+      });
```
Comment (Boobaley) [inline src/pages/CounterOfCounters/containers/CounterOfCountersContainer.js:22]: 1) Don't mutate entire data list
2) use `map` here.

File: src/pages/CounterOfCounters/containers/CounterOfCountersContainer.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+      return copyCounters;
+    });
+  }, []);
+
+  const handleResetCounters = useCallback(() => {
+    setCounters((counters) => {
+      const copyCounters = [...counters];
+      copyCounters.splice(0, copyCounters.length);
+      return copyCounters;
+    });
```
Comment (Boobaley) [inline src/pages/CounterOfCounters/containers/CounterOfCountersContainer.js]: `setCounters([])` have sense?

File: src/pages/CounterOfCounters/containers/CounterOfCountersContainer.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+  }, []);
+
+  const handleReset = useCallback((id) => {
+    setCounters((counters) => {
+      const copyCounters = [...counters];
+      copyCounters.forEach((counter) => {
+        if (counter.id === id) {
+          counter.countValue = 0;
+        }
+      });
```
Comment (Boobaley) [inline src/pages/CounterOfCounters/containers/CounterOfCountersContainer.js:77]: same as above. use `map` instead of `forEach`. fix it in all places

**Review by Boobaley — CHANGES_REQUESTED** [review]

File: src/pages/CounterOfCounters/components/CounterItem/index.js (hunk 1 above)
Comment (PavelPashkovich) [reply src/pages/CounterOfCounters/components/CounterItem/index.js]: Fixed

File: src/pages/CounterOfCounters/components/CountersLayout/index.js (hunk 2 above)
Comment (PavelPashkovich) [reply src/pages/CounterOfCounters/components/CountersLayout/index.js]: Fixed

File: src/pages/CounterOfCounters/containers/CounterOfCountersContainer.js (hunk 3 above)
Comment (PavelPashkovich) [reply src/pages/CounterOfCounters/containers/CounterOfCountersContainer.js:22]: Fixed

File: src/pages/CounterOfCounters/containers/CounterOfCountersContainer.js (hunk 5 above)
Comment (PavelPashkovich) [reply src/pages/CounterOfCounters/containers/CounterOfCountersContainer.js:77]: Fixed

File: src/pages/CounterOfCounters/containers/CounterOfCountersContainer.js (hunk 4 above)
Comment (PavelPashkovich) [reply src/pages/CounterOfCounters/containers/CounterOfCountersContainer.js]: Fixed

**Pull request merged**